ToolBox/TradingEconomicsDataDownloader/TradingEconomicsCalendarDownloader.cs

```diff
@@ -48,52 +55,126 @@ public TradingEconomicsCalendarDownloader(string destinationFolder)
         /// <returns>True if process all downloads successfully</returns>
         public override bool Run()
         {
+            Log.Trace("TradingEconomicsCalendarDownloader.Run(): Begin downloading calendar data");
+
+            // Create the destination directory so that we don't error out in case there's no data
+            Directory.CreateDirectory(Path.Combine(_destinationFolder, "calendar"));
+
             var stopwatch = Stopwatch.StartNew();
             var data = new List<TradingEconomicsCalendar>();
+            var availableFiles = Directory.GetFiles(Path.Combine(_destinationFolder, "calendar"), "*.zip", SearchOption.AllDirectories)
+                .Where(
```

Comment:
I think this is parsing the `date` twice? Maybe could `Select` first, something like:
```suggestion
                var availableFiles = Directory.GetFiles(Path.Combine(_destinationFolder, "calendar"), "*.zip", SearchOption.AllDirectories)
                .Select(files =>
                    {
                        try
                        {

                            return DateTime.ParseExact(
                                    Path.GetFileName(files).Substring(0, 8),
                                    "yyyyMMdd",
                                    CultureInfo.InvariantCulture);
                        }
                        catch (Exception)
                        {
                            return DateTime.MaxValue;
                        }
                    }
                )
                .ToHashSet(time => time != DateTime.MaxValue);
```